### DISCUSSION TITLE
Resources when mirroring all IRRs?

### DISCUSSION THREAD

**Discussion by stkonst** [irrdnet/irrd#669] [General]
Hi Sasha/all,

I was wondering how much resources do I need to provide to my VM if I want to mirror all major IRRs (RIPE,ARIN,LACNIC,APNIC,AFRINIC) and have IRRd running stable?

I believe that the recommended resources listed in the documentation are not enough. Especially the RAM size and the disk space required for the postgress. 

Did any member of the community have a successful deployment with all major IRRs that can share the specs of his node? 

Kind Regards
Stavros

**Comment by troy2914** [comment]
We run with a few more IRRs sources than you have listed, and with RPKI enabled on most.
we configure our VMs with:
CPU     32 CPU(s)
Memory  192 GB
Though those are likely overkill for most cases.  And we are running irrd 4.2.4

**Comment by job** [comment]
I'm running IRRd 4.2.5 with 4 cores, 32 GB RAM, 512 GB Samsung 840 SSD with these databases:

```
irr$ echo '!j-*' | nc irr.yycix.ca 43
A264
RPKI:N:-
ARIN:Y:0-358693
RIPE:Y:0-53859417
APNIC:Y:0-10232537
LACNIC:Y:0-422200
AFRINIC:Y:0-1216604
NTTCOM:Y:0-1297659
RADB:Y:0-13644487
CANARIE:Y:0-4470
LEVEL3:Y:0-835712
ALTDB:Y:0-95297
JPIRR:Y:0-222710
IDNIC:Y:0-199263
BELL:Y:0-249600
RIPE-NONAUTH:Y:0-53859417
```

**Comment by fischerdouglas** [comment]
😭😭😭
Where is TC?
😭😭😭

Em dom., 24 de jul. de 2022 09:22, Job Snijders ***@***.***>
escreveu:

**Comment by mxsasha** [comment]
I run a main branch instance with NTTCOM,RADB,RIPE,RIPE-NONAUTH,ALTDB,BELL,LEVEL3,RGNET,APNIC,JPIRR,ARIN,BBOI,TC,AFRINIC,RPKI,PANIX,OPENFACE,NESTEGG,CANARIE,LACNIC with RPKI enabled for all, on [this quad core celeron](https://www.hardkernel.com/shop/odroid-h2plus/), 32GB ram, some scavenged SSD. I also use it for a few other things. My /var/lib/postgres is 30GB. I also have a 4.2 and 4.1 instance on 8GB each.

I have 5 HTTP and 5 whois workers running (each of these needs ~200MB). Initial imports also need more memory, so you could probably limit peak use by not adding all the large ones at the same time. On the 8GB nodes this is something I have to do if I need to run a full import of multiple sources.

**Comment by stkonst** [comment]
Thank's all for the replies. Based on your feedback I see that disk space and RAM are the most important resources for the IRRdv4 daemon.

@mxsasha  indeed when I bootstrap IRRd (with recommended resources) it consumes significantly more resources and I need to constantly feed the VM with memory and disk. However, after a week it stabilizes and resource consumption remains at the same levels.
